packages/angular_devkit/build_angular/src/angular-cli-files/plugins/single-test-transform.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { logging } from '@angular-devkit/core';
+import { getOptions } from 'loader-utils';
+import { loader } from 'webpack';
+
+export interface SingleTestTransformLoaderOptions {
+  files: string[];
+  logger: logging.Logger;
+}
+
+export default function loader(this: loader.LoaderContext, source: string) {
+  const options = getOptions(this) as SingleTestTransformLoaderOptions;
+  const start = 'import \'';
+  const end = '\';';
+  const testCode = start + options.files
```

Comment:
Actually, I think these can't be static `import` statements, because `import` statements will load their path before loading the current module.

Instead these need to be `require()` calls so that they are dynamic and happen at this particular point, after the setup in the beginning of the test file.

Reply:
I don't think the 2nd comment is a problem, this is the transformed `test.ts`:
![image](https://user-images.githubusercontent.com/1985429/58505014-3e2c3580-818c-11e9-8bf6-7257cc1d6449.png)

Aside of that, I've replaced `${path.replace('.ts', '')`, with `${path.replace('.' + extname(path), '')` to be on the safe side (extname is imported from 'path')

Reply:
I really thing the dynamic `require` are necessary. This is what I mean: https://stackoverflow.com/a/29334761/2116927

So even though the `import` is after `getTestBed()` (the important bit of initialization), it will run before it.

We don't have to assume this though. We can test it. 

- `npm i -g http-server`
- create these files
```
// index.html
<script src="./main.js" type="module"></script>

// main.js
console.log('before importing another-module');
import './another-module.js';
console.log('after importing another-module');

// another-module.js
console.log('inside another-module');
```
- run `http-server`
- open http://localhost:8080/, see console
```
another-module.js:1 inside another-module
main.js:1 before importing another-module
main.js:3 after importing another-module
```

You can see that the content of `another-module.js` was evaluated before the statement preceding the import statement.

In contrast, `require` statements are dynamic and sync, and will occur in that particular place. This of course is dependent on how Webpack implements the semantics, but as long as we stick to the intended semantics then it should be fine (and if it isn't, then it's a bug somewhere else).

Reply:
hm, I did not expect behavior like this - Looks like I've learnt something new today. Thanks!

Aside of that, I've changed to use require instead of import when transforming test.ts. Still works :)